File: update_currency.dart
import 'dart:io';

void main() {
  final dir = Directory('mobile/lib');
  final files = dir.listSync(recursive: true).whereType<File>().where((f) => f.path.endsWith('.dart'));

  for (final file in files) {
    String content = file.readAsStringSync();
    bool changed = false;

    // 1. Replace -\$${...} with -${...} EGP
    if (content.contains(r'-\$${')) {
      content = content.replaceAllMapped(RegExp(r'-\$\$\{([^}]+)\}'), (match) {
        return '-\${${match.group(1)}} EGP';
      });
      changed = true;
    }

    // 2. Replace +\$${...} with +${...} EGP
    if (content.contains(r'+\$${')) {
      content = content.replaceAllMapped(RegExp(r'\+\$\$\{([^}]+)\}'), (match) {
        return '+\${${match.group(1)}} EGP';
      });
      changed = true;
    }

    // 3. Replace \$${...} with ${...} EGP
    if (content.contains(r'\$${')) {
      content = content.replaceAllMapped(RegExp(r'\$\$\{([^}]+)\}'), (match) {
        return '\${${match.group(1)}} EGP';
      });
      changed = true;
    }

    // 4. Replace literal \$45 with 45 EGP
    if (content.contains(r'\$')) {
      final literalRegex = RegExp(r'\$(\d+(?:,\d+)?(?:\.\d+)?)');
      if (literalRegex.hasMatch(content)) {
        content = content.replaceAllMapped(literalRegex, (match) {
          return '${match.group(1)} EGP';
        });
        changed = true;
      }
    }

    if (changed) {
      file.writeAsStringSync(content);
      print('Updated: \${file.path}');
    }
  }
}

File: update_currency.py
import os
import re

def replace_currency(file_path):
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()

    # Replacements
    # \$${ -> EGP ${
    content = content.replace(r'\$${', r'EGP ${')
    # \$$ -> EGP $
    content = content.replace(r'\$$', r'EGP $')
    # \$ -> EGP  (only if followed by a number or ${)
    content = re.sub(r'\\\$(\d+)', r'EGP \1', content)
    
    # Replace any literal 'USD' with 'EGP' (be careful with variables, maybe just in strings)
    # Actually, we did 'USD' default in helpers.dart already, let's just do it directly.
    
    with open(file_path, 'w', encoding='utf-8') as f:
        f.write(content)

def main():
    lib_dir = os.path.join('mobile', 'lib')
    for root, dirs, files in os.walk(lib_dir):
        for file in files:
            if file.endswith('.dart'):
                replace_currency(os.path.join(root, file))

if __name__ == '__main__':
    main()
